### ISSUE TITLE
[Carp Serializable][Feature] Allow specifying how to handle serialization errors due to unregistered class type

### ISSUE THREAD

**Issue by andreasmpet** [cph-cachet/carp.sensing-flutter#414]
Hi! I'm using your carp_serializable library to deserialize json fetched from a Sanity-based backend, and would like for my app to handle new unregistered types gracefully. 

That is, I want that if the backend suddenly returns a json like this:

```
{
"__type": "newUnsupportedTypeFromBackend",
...<other data>
}
```

I could define a closure that returned an arbitrary object if the type is not found. 

To me, this is preferable to the whole json-deserialization failing because the backend suddenly decided to introduce a new type, and I haven't had the time to add support for it yet.

**Comment by bardram** [comment]
I've released version 2.0 of the package which should address this issue, using this construct:

```dart
@JsonSerializable(includeIfNull: false)
class B extends A {
  String? str;

  B([super.index, this.str]) : super();

  @override
  String get jsonType => 'dk.carp.$runtimeType';

  @override
  Function get fromJsonFunction => _$BFromJson;
  factory B.fromJson(Map<String, dynamic> json) =>
      FromJsonFactory().fromJson<B>(json, notAvailable: B(-1));
  @override
  Map<String, dynamic> toJson() => _$BToJson(this);
}
```
that specifies a `notAvailable` parameter that is used in the case an uknown type is encountered in the `json`.

**Comment by andreasmpet** [comment]
Amazing! Thank you so much for the quick turnaround!
